devicemodel/hw/pci/core.c

```diff
@@ -564,7 +565,12 @@ pci_emul_alloc_resource(uint64_t *baseptr, uint64_t limit, uint64_t size,
 		return -1;
 	}
 
-	base = roundup2(*baseptr, size);
+	/* PCI spec said that BAR base should be naturally aligned. On ACRN
+	 * if the bar size < PAGE_SIZE, BAR base should be aligned with
+	 * PAGE_SIZE. This is because the minimal size that EPT can map/unmap
+	 * is PAGE_SIZE.
+	 */
```

Comment:
Should we simply replace base with PAGE_SIZE in this case?

Usually it means the space [size, PAGE_SIZE] is simply wasted.
If we use that space for other purpose, we may have trouble.

Reply:
Hi Eddie, 
Do you mean we just use PAGE_SIZE for this ivshmem case?

Yes, more MMIO address space will be wasted if we have many BARs whose size is < 4KB.
Currently all the BAR addresses are allocated uniformly in device model. 
When trying to allocate an address region for a new BAR, those gap used for alignment is just skipped and will not be used.

Reply:
I mean adjust size like following before Line 567:
 If (size < PAGE_SIZE)
    size = PAGE_SIZE;

Reply:
Thanks for pointing this out.
Will submit a new PR.